### PR TITLE
`azurerm_subscription` - Replace Track1 with Pandora SDK

### PR DESCRIPTION
**Pull request by aristosvo**
Running Acceptance Tests is kind of impossible for me at the moment as I don't have an EA account which I can play with a bit more.

- [ ] I'll add some tests to make sure it _can_ be validated what I did.

### CODE REVIEW

File: go.mod
Comment (tombuildsstuff) [inline go.mod:16]: just an FYI since `go-azure-sdk` tends to get updated pretty frequently, we tend to pull the SDK upgrade out into it's own PR (for example I've done this in #18833, but hadn't noticed this PR) - perhaps it'd be worth pulling the SDK upgrade out of this/rebasing this on top of #18833, so that we can merge that and focus on the SDK-switch here rather than the SDK upgrade?

Comment (aristosvo) [reply go.mod:16]: Sure! I did it here as I suspected one of the HC maintainers would pull in a new version quit soon and I can easily rebase my PR on it after that being merged.

Changes within the `go-azure-SDK` are coming in so often, hard to keep that in sync:)

File: internal/services/subscription/subscription_resource.go
Comment (manicminer) [inline internal/services/subscription/subscription_resource.go:102]: Since this field is present in the old model, I suspect there's a good reason we didn't include it in this resource. There is also the `azurerm_management_group_subscription_association` resource and the `subscription_ids` property on `azurerm_management_group`, so perhaps we should leave this out for now?

(If we _do_ keep it, we'll need some additional test coverage and docs)

**Review by manicminer — CHANGES_REQUESTED** [review]
Hi @aristosvo, this looks great and the tests are passing. I have reservations about adding the `management_group_id` property here though, more details inline. If you can take a look then this should be good to merge. Thanks!

**Comment by aristosvo** [comment]
Rebased and management group removed

**Comment by jackofallops** [comment]
Hi @aristosvo - Hope you don't mind, I'm digging into an API permissions issue on this resource and I've pushed a few changes. I may need to do more depending on the outcome of the investigation. For context, permissions that allow a user to create subscriptions in the Portal are insufficient via the API. (at least for MCAs) As soon as that's resolved, and we can test this works as expected, we can get it merged.

**Comment by jackofallops** [comment]
Tested manually:
```
Plan: 1 to add, 0 to change, 0 to destroy.
azurerm_subscription.test: Creating...
azurerm_subscription.test: Still creating... [10s elapsed]
azurerm_subscription.test: Still creating... [20s elapsed]
azurerm_subscription.test: Still creating... [30s elapsed]
azurerm_subscription.test: Still creating... [40s elapsed]
azurerm_subscription.test: Still creating... [50s elapsed]
azurerm_subscription.test: Still creating... [1m0s elapsed]
azurerm_subscription.test: Still creating... [1m10s elapsed]
azurerm_subscription.test: Still creating... [1m20s elapsed]
azurerm_subscription.test: Still creating... [1m30s elapsed]
azurerm_subscription.test: Creation complete after 1m38s [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```
image


```
Plan: 0 to add, 0 to change, 1 to destroy.
azurerm_subscription.test: Destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 1m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 2m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 3m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 4m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 5m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 6m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 7m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 8m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 9m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 10m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m30s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m40s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 11m50s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 12m0s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 12m10s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 12m20s elapsed]
azurerm_subscription.test: Still destroying... [id=/providers/Microsoft.Subscription/aliases/testAcc-SteDev20230329-3, 12m30s elapsed]
azurerm_subscription.test: Destruction complete after 12m31s

Destroy complete! Resources: 1 destroyed.
```
image

**Review by katbyte — APPROVED** [review]
LGTM 🪗

**Pull request merged**
